### PR TITLE
Responsive comic

### LINKED ISSUE #7652: Comic image gallery component for multi-frame comics
# What is it and who's it for?
https://wellcome.slack.com/archives/C8X9YKM5X/p1644591471387349

As our webcomics begins to include more multi-frame comics, the image gallery component is not able to make best use of responsive design to keep the individual frames as large and readable as possible on different device screens.

Example:
https://preview.wellcomecollection.org/articles/YgP2JxMAACAAKVB5

Ideally component would be able to take a series of single frames and auto-grid then depending on output device.

## Other considerations 
<---Delete as appropriate--->
- Should/does this just affect comics?

### PR DESCRIPTION
**Pull request by davidpmccormick**
Part of #8400
Closes #7652

## Who is this for?
People who want comics to be more readable at more screen sizes

## What is it doing for them?
Allowing comics to be split into frames that break responsively

If a comic can easily be split up into frames, then content editors can use the `frames` label in the `ImageGallery` component (rather than the `standalone`) label they use currently. If for any reason it can't, then the `standalone` option continues to exist.

Up till now, we've made things in the labels conform to a notion of 'weight', but the new label, 'frames', doesn't really fit into that categorisation. I'm not convinced that the 'weight' categorisation has ever really quite worked in practice as it was meant to in theory, and perhaps it's time to let it go?

I haven't been able to readily emulate the current image-sizing logic we have where the single comic image is constrained by the height of the viewport. I've opted to instead to limit things that are likely to be more landscape to 10 columns on desktop and anything that is likely to be more square/portrait to 8 columns on desktop. This felt ok to me when testing it. If it turns out not to be ok, I don't think it'd be an insurmountable challenge to figure it out, but I didn't want to drop any more time on it unnecessarily.

### CODE REVIEW

**Review by alexwlchan — APPROVED** [review]
This looks fine; the weight stuff is a bit odd but I don't think needs to be fixed right now – maybe open a ticket to come back to it later?

(Sorry, I keep nearly reviewing this and then not clicking the button 😬 )

File: content/webapp/components/ImageGallery/ImageGallery.tsx [hunk 1]
```diff
@@ -201,6 +242,7 @@ const ImageGallery: FunctionComponent<{ id: number } & Props> = ({
   title,
   items,
   isStandalone,
+  isFrames,
```
[the remark this answers is not in the document]
Comment (davidpmccormick) [reply content/webapp/components/ImageGallery/ImageGallery.tsx:245]: Yeah, I think this weight stuff breaking semantically is a bit odd. The only other gallery type is the default, unlabelled type.

This is the current logic

No label: show first image and include button to reveal all subsequent images
Standalone: always show all images (used for photo essays and comic currently)
Frames: show all images responsively (comic from now on)

[the remark this answers is not in the document]
Comment (alexwlchan) [reply content/webapp/components/ImageGallery/ImageGallery.tsx:245]: Is this a field that gets set in Prismic? If so, I wonder if we rename `frames` to `comics` to make it more obvious why it should be used?

[the remark this answers is not in the document]
Comment (davidpmccormick) [reply content/webapp/components/ImageGallery/ImageGallery.tsx:245]: I remain unconvinced that the weight metaphor has every really worked well for us – more often than not I have to check what 'standalone' and 'featured' actually mean as applied to a given slice-zone. I suppose it aims to limit the proliferation of styles, but I think it does so at the risk of hamstringing us a bit

[the remark this answers is not in the document]
Comment (davidpmccormick) [reply content/webapp/components/ImageGallery/ImageGallery.tsx:245]: We could do, and I take your point – the only thing I'm thinking is that I was talking to Ben and he was expressing an interest in being able to art-direct other images in a similar way, in which case frames would continue to be appropriate. But perhaps that's unrealistic. I don't think 'frames' is too abstract though?

[the remark this answers is not in the document]
Comment (alexwlchan) [reply content/webapp/components/ImageGallery/ImageGallery.tsx:245]: I reckon it's fine as long as it makes sense to Ben, and we table a broader discussion of how to handle `weight` for another time.

**Pull request merged**